### PR TITLE
feat: add support for common labels in helm charts

### PR DESCRIPTION
**Pull request by thib-mary**
Usefull when we have a policies manager as Kyverno.

Signed-off-by: Thibault <mary.thibault2@gmail.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

When we used a policies manager as Kyverno, it's usefull to be able to add a common labels on all resources created by the chart in order to respect the policy.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:

N/A

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Comment by nilekhc** [comment]
Hi @thib-mary, Thanks for the PR. Could you move these changes to [manifest_staging](https://github.com/kubernetes-sigs/secrets-store-csi-driver/tree/main/manifest_staging)? We promote them to charts while releasing the driver.

File: manifest_staging/charts/secrets-store-csi-driver/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: secrets-store-csi-driver
-version: 1.2.2
-appVersion: 1.2.2
+version: 1.3.2
+appVersion: 1.3.2
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/Chart.yaml:4]: Can you revert this change? We will update the version while releasing the chart

File: manifest_staging/charts/secrets-store-csi-driver/values.yaml [hunk 2]
```diff
@@ -217,3 +217,7 @@ imagePullSecrets: []
 tokenRequests: []
 # - audience: aud1
 # - audience: aud2
+
+# -- Labels to apply to all resources
+commonLabels: {}
+# team_name: dev
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/values.yaml:223]: nit: add newline

File: manifest_staging/charts/secrets-store-csi-driver/templates/_helpers.tpl [hunk 3]
```diff
@@ -35,6 +35,9 @@ labels:
   app.kubernetes.io/version: "{{ .Chart.AppVersion }}"
   app: {{ template "sscd.name" . }}
   helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+  {{- if .Values.commonLabels}}
+  {{ toYaml .Values.commonLabels }}
+  {{- end }}
```
Comment (nilekhc) [inline manifest_staging/charts/secrets-store-csi-driver/templates/_helpers.tpl:40]: @thib-mary fyi `sscd.labels` is not added in all templates.

Comment (thib-mary) [reply manifest_staging/charts/secrets-store-csi-driver/templates/_helpers.tpl:40]: > @thib-mary fyi `sscd.labels` is not added in all templates.

Indeed, can I still update? @nilekhc

Comment (nilekhc) [reply manifest_staging/charts/secrets-store-csi-driver/templates/_helpers.tpl]: Sure. I noted this to make sure it satisfies your use case.

File: manifest_staging/charts/secrets-store-csi-driver/templates/crds-upgrade-hook.yaml [hunk 4]
```diff
@@ -2,7 +2,8 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: {{ template "sscd.fullname" . }}-upgrade-crds
-{{ include "sscd.labels" . | indent 2 }}
+  labels:
+    {{ include "sscd.labels" . | indent 4 }}
```
Comment (aramase) [inline manifest_staging/charts/secrets-store-csi-driver/templates/crds-upgrade-hook.yaml:6]: This format needs to be updated in all the other resources in the file and `templates/*` directory

**Review by aramase — CHANGES_REQUESTED** [review]

**Comment by aramase** [comment]
/retitle feat: add support for common labels in helm charts

File: manifest_staging/charts/secrets-store-csi-driver/templates/crds-upgrade-hook.yaml (hunk 4 above)
Comment (thib-mary) [reply manifest_staging/charts/secrets-store-csi-driver/templates/crds-upgrade-hook.yaml:6]: > This format needs to be updated in all the other resources in the file and `templates/*` directory

Hum, it already is, isn't it ? @aramase

Comment (aramase) [reply manifest_staging/charts/secrets-store-csi-driver/templates/crds-upgrade-hook.yaml:6]: @thib-mary Looking at the `crds-upgrade-hook.yaml` file: https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/acbb36adf6ba8be8d3c8994828dbd7e528cbe43c/manifest_staging/charts/secrets-store-csi-driver/templates/keep-crds-upgrade-hook.yaml, I don't see the change for all other resources.

**Review by aramase — COMMENTED** [review]
Thanks for the PR, @thib-mary!

/lgtm
/approve

/hold cancel
